frontend/src/main/web/src/components/Flex/index.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { PropTypes } from 'react' // eslint-disable-line
+import { merge } from 'lodash'
+import {
+  Base
+} from '../'
+
+const classes = {
+  base: {
+    d: 'D(f)'
+  },
+  align: {
+    fs: 'Ai(fs)',
+    fe: 'Ai(fe)',
+    c: 'Ai(c)',
+    b: 'Ai(b)',
+    st: 'Ai(st)'
+  },
+  alignContent: {
+    fs: 'Ac(fs)',
+    fe: 'Ac(fe)',
+    c: 'Ac(c)',
+    st: 'Ac(st)',
+    sb: 'Ac(sb)',
+    sa: 'Ac(sa)'
+  },
+  dir: {
+    r: 'Fld(r)',
+    rr: 'Fld(rr)',
+    c: 'Fld(c)',
+    cr: 'Fld(cr)'
+  },
+  justify: {
+    fs: 'Jc(fs)',
+    fe: 'Jc(fe)',
+    c: 'Jc(c)',
+    sb: 'Jc(sb)',
+    sa: 'Jc(sa)'
+  },
+  wrap: {
+    nw: 'Flw(nw)',
+    w: 'Flw(w)',
+    wr: 'Flw(wr)'
+  }
+}
+
+const Flex = ({
+  align,
+  alignContent,
+  children,
+  dir,
+  justify,
+  wrap,
+  theme = {},
+  ...props
+}) => {
+  const themed = merge({}, classes, theme)
+  const stateTheme = merge({},
+    themed.base,
+    {
+      ai: themed.align[align],
+      ac: themed.alignContent[alignContent],
+      fld: themed.dir[dir],
+      jc: themed.justify[justify],
+      flw: themed.wrap[wrap]
+    }
+  )
+  return (
+    <Base
+      componentName='Flex'
+      theme={stateTheme}
+      {...props}
+    >
+      {children}
+    </Base>
+  )
+}
+
+Flex.propTypes = {
```

Comment:
Not sure what you meant in here. What information you're referring to? the props?